data_override/data_override.F90

```diff
@@ -37,1267 +41,163 @@
 !! data_override will take place, field values outside the region will not be affected.
 
 module data_override_mod
-use yaml_parser_mod
-use constants_mod, only: PI
-use mpp_mod, only : mpp_error, FATAL, WARNING, NOTE, stdout, stdlog, mpp_max
-use mpp_mod, only : input_nml_file
-use horiz_interp_mod, only : horiz_interp_init, horiz_interp_new, horiz_interp_type, &
-                             assignment(=)
-use time_interp_external2_mod, only:time_interp_external_init, &
-                                   time_interp_external, &
-                                   init_external_field, &
-                                   get_external_field_size, &
-                                   set_override_region, &
-                                   reset_src_data_region, &
-                                   NO_REGION, INSIDE_REGION, OUTSIDE_REGION,     &
-                                   get_external_fileobj
-use fms_mod, only: write_version_number, lowercase, check_nml_error
-use axis_utils2_mod,  only : nearest_index, axis_edges
-use mpp_domains_mod, only : domain2d, mpp_get_compute_domain, NULL_DOMAIN2D,operator(.NE.),operator(.EQ.)
-use mpp_domains_mod, only : mpp_get_global_domain, mpp_get_data_domain
-use mpp_domains_mod, only : domainUG, mpp_pass_SG_to_UG, mpp_get_UG_SG_domain, NULL_DOMAINUG
-use time_manager_mod, only: time_type
-use fms2_io_mod,     only : FmsNetcdfFile_t, open_file, close_file, &
-                            read_data, fms2_io_init, variable_exists, &
-                            get_mosaic_tile_file
-use get_grid_version_mod, only: get_grid_version_1, get_grid_version_2
+  use data_override_r4
+  use data_override_r8
```

Comment:
So all the derived types are private and named the same within these two modules right?

Reply:
That's correct - only the public subroutines have different `r4`/`r8` names.